dbschema/src/main/java/org/folio/rest/persist/ddlgen/Index.java

```diff
@@ -83,53 +85,40 @@ public String getFinalSqlExpression(String tableLoc) {
         if(i != 0) {
           result .append(" , ");
         }
-        appendExpandedTerm(tableLoc, splitIndex[i], result);
+        result.append(formatExpandedTerm(tableLoc, splitIndex[i]));
       }
       result.append(")");
       return result.toString();
     }
 
   }
-  public static void appendExpandedTerm(String tableLoc, String  currentTerm, StringBuilder result) {
-    String [] rawExpandedTerm = currentTerm.split("\\.");
-    StringBuilder expandedTerm = formatExpandedTerm(tableLoc + ".jsonb",rawExpandedTerm);
-    result.append(expandedTerm);
-  }
 
-  public static StringBuilder formatExpandedTerm(String table, String[] rawExpandedTerm) {
+  public static StringBuilder formatExpandedTerm(String table, String term) {
     StringBuilder expandedTerm = new StringBuilder();
     StringBuilder result = new StringBuilder();
-    boolean wasArrayIndex = false;
-    expandedTerm.append(table);
-    for(int j = 0; j < rawExpandedTerm.length; j++) {
-      int idx = rawExpandedTerm[j].indexOf("[*]");
+
+      int idx = term.indexOf("[*]");
+      //case where the syntax is found
       if(idx > -1) {
-        wasArrayIndex = appendExpandedArrayTerm(rawExpandedTerm, expandedTerm, j, idx);
-        break;
+        expandedTerm.append(table).append(".").append(Cql2PgUtil.cqlNameAsSqlJson("jsonb",term.substring(0,idx)));
+
+        int arrayTermPresent = term.indexOf( "[*].");
+        if(arrayTermPresent > -1) {
+          formatArrayWithTerm(term, expandedTerm, result, arrayTermPresent);
+        } else {
+          result.append("concat_array_object(").append(expandedTerm);
+        }
+        result.append(")");
       } else {
-        wasArrayIndex = appendExpandedSimpleTerm(rawExpandedTerm, expandedTerm, j, idx);
+        result.append(table).append(".").append(Cql2PgUtil.cqlNameAsSqlText("jsonb",term));
       }
-    }
-    if(wasArrayIndex) {
-      result.append("concat_array_object_values(").append(expandedTerm).append(")");
-    } else {
-      result.append(expandedTerm);
-    }
     return result;
```

Comment:
Can you make the three cases more visible? Example:

```
// term does not contain "[*]"
if (idx == -1) {
  result.append(...
  return result;
}

// term ends with "[*]"
if (idx == term.length() - 3) {
  result.append(...
  return result;
}

// term ends with "[*].field"
result.append(...
return result;
```

The code in method formatArrayWithTerm is easy enough to move it into the calling method.